packages/babel-parser/src/plugins/estree.js

```diff
@@ -31,6 +33,24 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return node;
     }
 
+    estreeParseBigIntLiteral(value: any): N.Node {
+      let bigInt = null;
+      try {
+        bigInt = BigInt(value);
+      } catch (e) {
+        // https://github.com/estree/estree/blob/master/es2020.md#bigintliteral
+        // In environments that don't support BigInt, this value
+        // will be null.
+      }
+
+      // Espree creates a "Numeric" token for BigIntLiterals.
+      this.state.type = tt.num;
```

Comment:
Is this safe?

Reply:
Maybe it would be safer to overwrite the last element of the `this.state.tokens` array in the overwritten `parseLiteral` method?

Reply:
Oh, I like that! Will update.

Reply:
I hadn't realized we do ESTree token conversion in `babel-eslint` - will update doing that instead!